apps/remix-ide/src/app/tabs/runTab/contractDropdown.js

```diff
@@ -80,14 +81,42 @@ class ContractDropdownUI {
     window.localStorage.setItem(`ipfs/${this.exEnvironment}/${this.networkName}`, this.ipfsCheckedState)
   }
 
+  enableContractNames (enable) {
+    if (enable) {
+      this.selectContractNames.removeAttribute('disabled')
+      this.selectContractNames.setAttribute('title', 'Select contract for Deploy or At Address.')
+    } else {
+      this.selectContractNames.setAttribute('disabled', true)
+      if (this.loadType === 'sol') {
+        this.selectContractNames.setAttribute('title', '⚠ Select and compile *.sol file to deploy or access a contract.')
```

Comment:
`⚠ Select and compile *.sol file to deploy or access a contract.` if the user already compiled but switch to abi and then back to the solidity fine, that sounds weird to show him this msg as he actually already compiled.

Reply:
but he switched. the current file is defining the state everywhere. it wouldn't be reset if autocompile is on